ops/docker/Dockerfile.packages

```diff
@@ -17,68 +17,75 @@ FROM alpine:3.16 as manifests
 RUN apk add coreutils
 
 WORKDIR /tmp
-COPY yarn.lock .nvmrc package.json ./src/
+COPY pnpm-lock.yaml pnpm-workspace.yaml .nvmrc package.json ./src/
 COPY packages src/packages/
 RUN mkdir manifests && \
   cd src && \
   # copy package.json recursively
   find . -name 'package.json' | xargs cp --parents -t ../manifests/ && \
-  # yarn.lock
-  cp yarn.lock ../manifests/ && \
+  # pnpm-lock.yaml
+  cp pnpm-lock.yaml ../manifests/ && \
+  # pnpm-workspace.yaml
+  cp pnpm-workspace.yaml ../manifests/ && \
   # .nvmrc
   cp .nvmrc ../manifests/
 
 FROM ethereumoptimism/foundry:latest as foundry
-FROM node:16-alpine3.14 as base
+# bullseye-slim is debian based
+# we use it rather than alpien because it's not much
+# bigger and alpine is often missing packages for node applications
+# alpine is not officially supported by node.js
+FROM node:16.16.0-bullseye-slim as base
 
 # Base: install deps
-RUN apk --no-cache add curl \
+RUN apt-get update && apt-get install -y \
+  curl \
   jq \
   python3 \
   ca-certificates \
   git \
+  g++ \
   make \
   gcc \
+  pkg-config \
   musl-dev \
-  linux-headers \
   bash \
-  build-base \
-  gcompat
+  # the following 4 deps are needed for node-hid
+  # which is a deep sub dependency of ethers to install
+  # correctly
+  pkg-config \
+  libusb-1.0-0-dev \
+  libudev-dev \
+  --no-install-recommends
```

Comment:
a very problematic dependency in my experience is node-hid which is a deep sub dependency of ethers.    Here these installations simply allow some postinstall scripts to run correctly.  yarn would allow these to fail silently